nibabel/testing/__init__.py

```diff
@@ -62,6 +65,22 @@ def assert_allclose_safely(a, b, match_nans=True, rtol=1e-5, atol=1e-8):
     assert_true(np.allclose(a, b, rtol=rtol, atol=atol))
 
 
+def check_iteration(iterable):
```

Comment:
I see you only assert that this is True in the tests.

How about just using `list(iterable)` in the tests instead, with a comment that this is to check whether the iterable can be iterated all the way through?